.github/workflows/rust.yml

```diff
@@ -23,7 +23,7 @@ jobs:
             ~/.cargo/registry
             target
           key: "${{runner.os}} Rust ${{steps.rust-toolchain.outputs.cachekey}} Lock ${{hashFiles('Cargo.lock')}}"
-      - run: cargo hack test --feature-powerset --at-least-one-of aws-lc-rs,ring --locked -- --include-ignored
+      - run: cargo hack test --feature-powerset --at-least-one-of aws-lc-rs,ring --mutually-exclusive-features aws-lc-rs,ring --locked -- --include-ignored
```

Comment:
I'm not a big fan of these mutually exclusive features. Since we technically support both aws-lc-rs and ring features being active at the same time (prefering ring in this case, just like hickory does), I think we should at least verify that our code compiles.

Reply:
so maybe an check step before test?

Reply:
That would probably be a good idea, since it should be irrelevant for the rest of the feature matrix